### PR TITLE
fix test runner so it works on windows.

### PR DESCRIPTION
**Pull request by lovettchris**
fix: make test runner call lean directly instead of through a bash shell script.  So this version of mdbook now implements the code snippet extraction in Rust instead of awk.

### CODE REVIEW

**Comment by lovettchris** [comment]
@Kha, love to get your input on this one...

**Comment by Kha** [comment]
Ah yes, I missed this one. I assume our docs pass using it?
I wanted to say that it may be worth trying to upstream this, but it looks like upstream is desperately short of maintainers.

File: src/book/mod.rs [hunk 1]
```diff
@@ -254,25 +303,28 @@ impl MDBook {
...
+                    fs::write(&path, snippet.block);
+
+                    let mut cmd = Command::new(test_binary);
+                    cmd.arg(&path);
+                    let output = cmd.output()?;
+
+                    if !output.status.success() {
+                        failed = true;
+                        error!(
+                            "./test returned an error:\n\
```
Comment (Kha) [inline src/book/mod.rs]: still mentions `./test`

**Comment by lovettchris** [comment]
> Ah yes, I missed this one. I assume our docs pass using it?

Yes it works and also fails if the code block contains an error.

image

File: src/book/mod.rs (hunk 1 above)
Comment (lovettchris) [reply src/book/mod.rs]: ah, thanks, fixed, it now uses test_binary, so the output is:

```
2021-09-29 14:12:07 [ERROR] (mdbook::book): lean returned an error:

--- stdout
C:\Users\clovett\AppData\Local\Temp\mdbook-gDsUzI\whatIsLean.md_11.lean:16:4: error: expected command

--- stderr
```

**Comment by Kha** [comment]
Thanks!

**Pull request merged**